File: frontend/lib/auth/models/login_response.dart
class LoginResponse {
  final String token;
  final UserInfo user;

  LoginResponse({required this.token, required this.user});

  factory LoginResponse.fromJson(Map<String, dynamic> json) {
    return LoginResponse(
      token: json['token'] as String,
      user: UserInfo.fromJson(json['user'] as Map<String, dynamic>),
    );
  }
}

class UserInfo {
  final String id;
  final String email;

  UserInfo({required this.id, required this.email});

  factory UserInfo.fromJson(Map<String, dynamic> json) {
    return UserInfo(
      id: json['id'] as String,
      email: json['email'] as String,
    );
  }
}

File: backend/src/main/java/top/kariscode/karisreview/auth/dto/LoginResponse.java
package top.kariscode.karisreview.auth.dto;

import java.util.UUID;

public class LoginResponse {

    private String token;
    private UserInfo user;

    public LoginResponse(String token, UUID id, String email) {
        this.token = token;
        this.user = new UserInfo(id, email);
    }

    public String getToken() { return token; }
    public void setToken(String token) { this.token = token; }
    public UserInfo getUser() { return user; }
    public void setUser(UserInfo user) { this.user = user; }

    public static class UserInfo {
        private UUID id;
        private String email;

        public UserInfo(UUID id, String email) {
            this.id = id;
            this.email = email;
        }

        public UUID getId() { return id; }
        public void setId(UUID id) { this.id = id; }
        public String getEmail() { return email; }
        public void setEmail(String email) { this.email = email; }
    }
}